snuba/datasets/entity.py

```diff
@@ -96,14 +96,19 @@ def get_all_join_relationships(self) -> Mapping[str, JoinRelationship]:
 
     def validate_required_conditions(
         self, query: Query, alias: Optional[str] = None
-    ) -> bool:
+    ) -> Optional[Set[str]]:
         if not self._required_filter_columns and not self._required_time_column:
-            return True
+            return None
 
         condition = query.get_condition_from_ast()
         top_level = get_first_level_and_conditions(condition) if condition else []
         if not top_level:
-            return False
+            missing = set()
+            if self._required_time_column:
+                missing.add(self._required_time_column)
+            if self._required_filter_columns:
+                missing |= set(self._required_filter_columns)
+            return missing
```

Comment:
Isn't this logic already covered between line 140 and 169 ?

Reply:
Not exactly, but I refactored this to be more complete anyways.